fetch/main.ts

```diff
@@ -0,0 +1,44 @@
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+async function fetchUsers(): Promise<User[]> {
+  try {
+    const response = await fetch('https://jsonplaceholder.typicode.com/users');
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    const data: User[] = await response.json();
+    console.log(data);
+  } catch (error) {
+    console.error('Error:', error);
+  }
+  return [];
```

Comment:
If you really want the function to return the data array then replace `return []` with `return data`... But you need to declare the variable on top of the function's code block: `let data = [];` and then replace line 14 with: `data = await response.json();`